broker/api/broker.yaml

```
@@ -0,0 +1,978 @@
+       ([RFC 7235](https://tools.ietf.org/html/rfc7235))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    Forbidden:
+      description: 'Forbidden
+       ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    NotFound:
+      description: 'Not found ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    InternalServerError:
+      description: 'Internal server error
+       ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    Error:
+      description: 'Unexpected error
+       ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+  schemas:
+    Build:
+      # type: object
+      description: Describes schema for build.
+      allOf:
+          - $ref: '#/components/schemas/BuildRegister' 
+          - type: object
+            properties:
+              status:
+                type: string
+                enum:
+                  - UNKNOWN
+                  - QUEUED
+                  - RUNNING
+                  - DELETING
+                  - SUCCEEDED
+                  - FAILED
+                example: QUEUED
+                description: Current condition/state of build. 
+              started_at:
+                type: string
+                format: date-time
+                example: 2021-06-11T17:32:28Z
+                description: Timestamp taken at the begining of build.
+              finished_at:
+                type: string
+                format: date-time
+                example: 2021-06-11T17:32:28Z
+                description: Timestamp taken when build is finished and ready to
+                 deploy.
+            required:
+              - status
+              - started_at
+              - finished_at
+    # BuildStates:
+    #   type: string
+    #   enum:
+    #     - UNKNOWN
+    #     - QUEUED
+    #     - RUNNING
+    #     - DELETING
+    #     - SUCCEEDED
+    #     - FAILED
+    BuildRegister:
+      type: object
+      description: Describes schema for registering build
+      properties:
+        images:
+          type: array
+          items: 
+            $ref: '#/components/schemas/Image'
+          description: List of image to be build. It can contain single or
+           multiple images depending on requirement of repository.
+        head_commit:
+          type: object
+          oneOf:
+            - $ref: '#/components/schemas/Branch'
+            - $ref: '#/components/schemas/Tag'
+          description: "Reference to the `commit` or `tag` which is to be used
```

```suggestion
          description: "Reference to the `commit` or `tag` which is to be used
          example: {"branch": "master", "commit_sha": "HEAD" }
```
So the example provided by swagger works